drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -544,20 +579,23 @@ local function do_configure(driver, device)
     return
   end
 
-  -- if not fingerprinted, dynamically configure base-lock profile based on Power Source cluster checks
-  local power_source_eps = device:get_endpoints(clusters.PowerSource.ID)
-  local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
+  -- if not fingerprinted, dynamically configure base-lock profile
   local profile_name = "base-lock"
-
-  -- check for battery type
-  if #power_source_eps == 0 then
+  device:set_field(PROFILE_BASE_NAME, profile_name, {persist = true})
+  local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
+  if #battery_feature_eps == 0 then
     profile_name = profile_name .. "-nobattery"
-  elseif #battery_feature_eps == 0 then
+  elseif not device:get_field(SUPPORT_BATTERY_PERCENTAGE) then
```

Comment:
I would not expect this value to have been set at the time of `doConfigure` since the `doConfigure` event itself sends the `read` request for the AttributeList below.

If #battery feature eps is 0, we could move forward with the metadata update to change the profile to the `no-battery` variety here, but if the #battery eps is > 0, then we could hold off on the metadata update so that the proper profile can be determined and then set in the AttributeList handler. That way, we could avoid setting the profile if we know it is likely to change when we receive an AttributeList report. What do you think?

Reply:
Good call! I changed it so that it would only update the profile if it finds that the battery feature is not available. The AttributeList handler can update it again once it runs.